Makefile.package

```diff
@@ -22,12 +22,15 @@ DATE := $(shell date +"%Y%m%d")
 TIMESTAMP := $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 
 HAS_GIT := $(shell [ -d ${PROJECT_ROOT}/.git ] && echo "true")
+ifeq (${PROJECT_VERSION},unknown)
 ifeq (${HAS_GIT},true)
-	GIT_COMMIT_HASH  := $(shell git log -1 --format=%h)
-	GIT_COMMIT_DIRTY := $(shell git diff --quiet || echo "-dirty")
-	GIT_DESCRIBE     := ${GIT_COMMIT_HASH}${GIT_COMMIT_DIRTY}
+GIT_COMMIT_HASH  := $(shell git log -1 --format=%h)
+GIT_COMMIT_DIRTY := $(shell git diff --quiet || echo "-dirty")
+GIT_DESCRIBE     := $(shell git describe --dirty="-dirty" | sed -r "s/^v(.*)/\1/")
```

Comment:
`git describe` may not work very well in CI context when shallow clones are used, but there is still the VERSION file which can be used instead.

Reply:
Precisely!